CRM/Contribute/BAO/Contribution/Utils.php

```diff
@@ -120,11 +120,6 @@ public static function processConfirm(
       $paymentParams['contributionTypeID'] = $contributionTypeId;
       $paymentParams['item_name'] = $form->_params['description'];
 
-      if ($contribution && $form->_values['is_recur'] && $contribution->contribution_recur_id
```

Comment:
I have tested recurring contributions, recurring memberships with additional amount and without additional amount. I still need to test with pricesets for both, but if every recurring transaction is created by the function processRecurringContribution, then I think the fix should be enough. I will however do more tests and run the api tests to ensure that it is indeed enough.